consensus/propagation/have_wants.go

```diff
@@ -221,6 +223,12 @@ func (blockProp *Reactor) requestFromPeer(ps *PeerState) {
 			if wants == nil {
 				continue
 			}
+			missingPartsCount := countRemainingParts(int(parts.Total()), len(parts.BitArray().GetTrueIndices()))
```

Comment:
same, but this takes into account the time it took to iterate over all the requests to create the want.